### PR TITLE
Debugger refactoring and cleanup

### PR DESCRIPTION
**Pull request by int19h**
Clean up the use of evaluation - ditch unnecessary helpers, don't use `BeginEvaluationAsync` where it's not necessary, do not tie `DebugEvaluationResult` to a stack frame.

Rename things so that they're more descriptive and/or not misleading.

Split types into separate source files for better readability.

Remove unused stuff.

Comments, comments everywhere!

### CODE REVIEW

**Comment by huguesv** [comment]
:+1:

File: src/Debugger/Impl/DebugEvaluationResult.cs [hunk 1]
```diff
@@ -2,113 +2,114 @@
...
 
-        public static string ToRVector(this DebugEvaluationResultFields fields) {
-            var fieldNames = _mapping.Where(kv => fields.HasFlag(kv.Key)).Select(kv => "'" + kv.Value + "'");
-            return Invariant($"base::c({string.Join(", ", fieldNames)})");
-        }
-    }
+        /// <summary>
+        /// R expression designating the environment in which the evaluation that produced this result took place.
+        /// </summary>
+        public string EnvironmentExpression { get; }
```
Comment (AlexanderSher) [inline src/Debugger/Impl/DebugEvaluationResult.cs]: I can't find where this property is used

Comment (int19h) [inline src/Debugger/Impl/DebugEvaluationResult.cs]: No-one uses it outside of the class (but I saw no reason to make it private). Inside the class, the first overload of `EvaluateAsync` uses it, as do `GetChildrenAsync` in `DebugValueEvaluationResult`.

**Comment by AlexanderSher** [comment]
:+1:

**Pull request merged**